### ISSUE TITLE
Project queries (copied from old repo)

### ISSUE THREAD

**Issue by krezac** [krezac/wireless-welding-pedal#1]
mr-sneezy commented 11 hours ago • 
Hi, I have a similar interest in a BT remote TIG welding pedal and have been considering hardware options...

Seeing as this is the start of your project, have you considered using an ESP32 module for the TX and RX ends instead of separate BT HM-10 and Arduino modules ?
Benefits are that it has an Arduino compatible MCU and BT & BLE (also WiFi) on the one module for around $5-$7USD.
The ESP32 with a MCP41010 10K digital potentiometer should work as the RX end...
Thoughts ?

@krezac
 
Owner
krezac commented 10 hours ago
Hi Martin,
I'm actually considering that, exactly for the reasons you mentioned (I just have the stuff mentioned in the description at home, so the plan was to use them for proof of concept). Application code should be more less the same anyway (virtual serial port over BT).

At least for TX, as there is a reasonably priced board which already includes battery charger - https://www.laskarduino.cz/lilygo-ttgo-t8-esp32-s2-v1-1-vyvojova-deska/

For RX, I'd probably go for "custom" (power supply + relay + digi pot) shield on top of sth like https://www.laskarduino.cz/mh-et-live-esp32-minikit--wemos-d1-mini-shield-kompatibilni/

(the links are just to some local shop, I'm quite sure these can be obtained anywhere)

What's not clear to me is if the regulated voltage on MCP digi pot is also limited to 5.5V (some welding machines use 0-10V, I need to check that for mine). Someone was recommending AD5290.

BTW this is the construction of another Czech guy, unfortunately no more details than these pictures survived. May be useful for mechanical construction, though.
https://louiss13.rajce.idnes.cz/TIG/

Regards

@mr-sneezy
 
Author
mr-sneezy commented 9 hours ago
Those two modules you mention are also listed on one of my favorite cheap parts buying sites (banggood.com), easy choice.

I'm waiting for a foot pedal (cable type) to arrive for my model TIG machine, but in the meantime I measure around 4V across my amp control socket pins.
My plan is to convert it to wireless with the BT modules.

I'll check data sheets about the MCP41010 limits tomorrow, yes better to use a chip that suits more machines.

**Comment by mr-sneezy** [comment]
Looking at the AD5290 data sheets, if we want the chip to control a welding machine that has 10V across it's control pot pins we need to supply the AD5390 with a VDD voltage equal or higher for it to work. That will complicate the circuit somewhat as we will need a 10V or high power rail from somewhere. If the machines have 5V or less like mine then VDD can be run from 5V same as the ESP32 (regulator input). Is that you understanding ?
Does you welding machine have more than 5V across it's amp control sense pins ?

**Comment by krezac** [comment]
Sorry for late response, I had to wait for a weekend (I don't have the welding stuff in my apartment but at my parents place).
I measured 5V on the connector so I personally don't need to care about 10V. I'll keep designing that shield for someone else ;).
ESP32 boards are on the way, now I need to order the pots.

**Comment by mr-sneezy** [comment]
That will keep it simple to start with. I already have some MCP41010 already and a spare ESP32, but I don't have a foot pedal though yet due to delivery delays. I could make the whole thing from scratch I guess, but it would be good to use a known good pedal to 'remote' it. 
It occurs to me that if BT serial is used then it's possible to generate test strings from an Android phone app like this to the RX device, before coding the TX device...
https://play.google.com/store/apps/details?id=de.kai_morich.serial_bluetooth_terminal

**Comment by krezac** [comment]
Yeah, the mechanical part will be the most challenging part of the project.
I've tried to 3d print something from thingiverse, but the the gears were all but smooth. So I think ver0 will be just two pieces of plywood, hinge and timing belt.

**Comment by krezac** [comment]
FYI (I did't have much time)
The modules arrived,  what I missed is that S2 chip on Liligo is not supported in Platformio (maybe some playing with libs will help as it works in Arduino IDE 2.0 alpha with alpha esspresstif). Doesn't matter at this point.
So far I've quick tried the esp-now combined with my POC code and deep sleep. It's functional, I need to measure the latency. 
I plan to do the same with BLE and with BT serial if I find the slave code for ESP32 somewhere (no success, but I haven't spent too much time on it).
Digi pots arrived today so if there is rainy weekend ahead, I may do some progress ;)

**Comment by mr-sneezy** [comment]
I did some reading about BLE and BT on the ESP32 also, as I know very little of BLE.
Seems BLE serial uses the same supply current as BT serial on the ESP32 anyway. So appears not much advantage in battery life to use BLE rather than BT serial.
If 18650 batteries are used at each end this is probably no issue, as you'd still get about 20 hours use out of a decent quality cell. 
 
I'm thinking latency would need to be less than 100mS or else it's going to feel laggy at the foot pedal ?

 I can try out code on a breadboard with a regular ESP32 Dev board. 
You are right about slave BT code,  I searched Github repo's for 'ESP32 bt slave' and got no results but did find a few hits with 'esp32 ble slave'.

**Comment by krezac** [comment]
My understanding is BLE uses different encoding on physical level so it should have longer range (in our case more noise resistant). But it's just a theory.

Deep sleep, it's not that much about battery life (there are more current eaters on the boards), I just don't like sending 0 every 50ms or so. So I wanted to see if the wake up is fast enough after pedal switch activation (I don't plan to sleep when pedal is active). Visually (LED) it looks fine, but I want to see some numbers.

Latency, I'd say the lower the  better (as there will be another delay in the machine sampling the pot, I assume). I was thinking about sending sth like 20 updates per second, 10 may be enough (it needs a real life test).

Reg testing, feel free. I'll push my sketches in the evening once I get to my private computer. But it's more less this code: https://randomnerdtutorials.com/esp-now-esp32-arduino-ide/ 
I plan to use the BLE code from this page as starting point for the proof of concept as well.

Yeah, maybe combining these two would be the way to go.

**Comment by mr-sneezy** [comment]
I'm now reading about ESP-Now, I was not aware of it before (or maybe forgot). Seems very useful for this sort of communication.

20Hz updates I think is a great target if possible.

My eyesight is not great for reading numbers down at the pedal on the floor (I'd be wearing reading glasses for welding that have short depth of field). I was thinking a single Neo-pixel LED or RGB LED would be a nice thing though, full colour range can indicate a lot, along with flashing modes.

I'll keep an eye out for your code.

**Comment by krezac** [comment]
I don't plan any display at the pedal (at the end it's under your foot). Battery-low LED next to the power switch maybe. But  even that one makes more sense on receiver side (sending one packet about low batt and falling in deep sleep) where it's more in sight than on the pedal somewhere on the floor.
The Neopixel sounds like a good idea.

**Comment by krezac** [comment]
https://github.com/krezac/wireless-welding-pedal/tree/main/firmware/poc

**Comment by mr-sneezy** [comment]
Saw the code. 
I understand from some reading docs, that ESP-Now on ESP32 is not communications compatible with ESP-Now used on ESP8266. 
So I've just ordered a couple more ESP32 and a couple of 0.9" OLED displays to test with. Will take a couple of weeks...

**Comment by mr-sneezy** [comment]
The OEM pedal arrived this week so I did some voltage probing with it installed on the TIG. I now measure up to 12V on the pot wiper pin to ground at maximum pedal, and about 0.3V with the pedal at zero (foot off).  
Not sure why I got 5V before, maybe I did not max out the current setting dial when I measured direct on the machine socket, as middle dial is around 6V. This will complicate a circuit for the RX end...

PS. I really don't like the ergonomic feel of the pedal that arrived, it sits quite high off the floor, feels very unnatural. I'll give some thought to making a better one before I go much further.

**Comment by krezac** [comment]
Well, that sucks.
I was playing with BLE a bit, it looks like the reconnect after deep sleep is slower a bit, but on the other side, the chips are much colder than in esp-now mode.

**Comment by DigitalScene3** [comment]
Hello gentlemen, i'm wanting to build my own wireless remote for both a Lincoln(invertec, vantage) & Miller(xmt350, big blue). But i was wondering how in the heck would i program them. So my question is, would this code you have here be able to connect the transmitting unit to the receiving control via Bluetooth and if so how would i go about setting the correct  parameters for amperage? (hopefully I'm saying all this correct since I've only been researching for about 3 days now). And if not could any of you point me in the right direction? Thank you.. also did you ever get your foot pedal to be wireless?

**Comment by krezac** [comment]
Hi @DigitalScene3 , from my side, the project is on hold. So the status is exactly what you see here. Maybe during winter, there will be some progress.

**Comment by DigitalScene3** [comment]
OK thanks.. I'll come back to check from time to time as well do some looking around for parts..
